statemachine_test.go

```diff
@@ -233,6 +236,33 @@ func TestStateMachine_DiscardGarbage(t *testing.T) {
 	)
 }
 
+type stateMachineTest struct {
+	run []string
+}
+
+func (sm *stateMachineTest) ActionA(t *T) { sm.run = append(sm.run, "actionA") }
+func (sm *stateMachineTest) ActionB(t TB) { sm.run = append(sm.run, "actionB") }
+func (sm *stateMachineTest) ActionC(t *T) { sm.run = append(sm.run, "actionC") }
+func (sm *stateMachineTest) Check(*T)     {}
+
+func TestStateMachineActions(t *testing.T) {
+	Check(t, func(rt *T) {
```

Comment:
This test does not check that the `TB` actions actually work properly. I'd say all it effectively does is checks that `ActionB` is present in the result. Better to just make a regular test that asserts that.

Also, you should not call `testing.T.Fatalf` from a rapid test. Naming an argument just `t` to shadow `t *testing.T` variable instead of e.g. `rt` is the preferred naming convention.

Reply:
Fixed the shadowing, and rewrote the test to use `t.Skip` to verify some functionality of the passed in `TB`.

I also added an additional test to verify the `TB` action can be used directly (more of a sanity test to ensure the type of the action is the interface)